src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -190,15 +190,30 @@ public void shouldAvoidFetchingReadConnectionWhenNotNecessary() throws SQLExcept
     }
 
     @Test
-    public void shouldRunNativeSqlOnMain() throws SQLException {
+    public void shouldRunNativeSqlOnReplica() throws SQLException {
         final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
         final Connection connection = DualConnection.builder(
             connectionProvider,
-            permanentInconsistency().build()
+            permanentConsistency().build()
         ).build();
 
         connection.nativeSQL(SIMPLE_QUERY);
 
+        assertThat(connectionProvider.getProvidedConnectionTypes())
+            .containsExactly(REPLICA);
+    }
+
+    @Test
+    public void shouldBePossibleToDisableNativeSqlFix() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).compatibleWithPreviousVersion()
```

Comment:
Looking from customer perspective, isn't this easier to introduce a bug for a person not familiar with `safe mode`? 
So some dev will just bump the version. How we can force him to think about possibility to introduce bugs?

One way would be to force `dual-connection` consumer to define : 
`compatibleWithPreviousVersion()` or `inompatibleWithPreviousVersion()`

Reply:
@afaruga-atlassian Please keep in mind it's an open-source library. Not all the products use feature flags, and not all the products are interested in the feature. By forcing people to use it, we make the library more complicated to use. `inompatibleWithPreviousVersion` is what every library do, and it's a good default.